Wikipedia/Code/WMFAnnouncementsContentSource.m

```diff
@@ -123,6 +126,33 @@ - (void)updateVisibilityOfNotificationAnnouncementsInManagedObjectContext:(NSMan
     }
 }
 
+- (void)saveNotificationsGroupInManagedObjectContext:(NSManagedObjectContext *)moc date:(NSDate *)date {
+    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+    BOOL isLoggedIn = self.fetcher.session.isAuthenticated;
+
+    if (userDefaults.wmf_shouldShowNotificationsExploreFeedCard) {
+
+        WMFContentGroup *currentNotificationsCardGroup = [moc newestGroupOfKind:WMFContentGroupKindNotification];
+
+        if (isLoggedIn) {
+            if (currentNotificationsCardGroup) {
+                if (!currentNotificationsCardGroup.isVisible && !currentNotificationsCardGroup.wasDismissed) {
+                    currentNotificationsCardGroup.isVisible = YES;
+                }
+            } else {
+                WMFContentGroup *newNotificationsCardGroup = [moc createGroupOfKind:WMFContentGroupKindNotification forDate:[NSDate date] withSiteURL:self.siteURL associatedContent:nil];
```

Comment:
My mistake on this one - we should probably just use the `date` parameter here instead of creating a new `[NSDate date]` here:

```suggestion
                WMFContentGroup *newNotificationsCardGroup = [moc createGroupOfKind:WMFContentGroupKindNotification forDate:date withSiteURL:self.siteURL associatedContent:nil];
```